functional-tests/src/test/groovy/FrontendCore/modules/Tab_Permit.groovy

```diff
@@ -3,13 +3,39 @@ package modules
 import geb.Module 
 
 class Tab_Permit extends Module {
+    static PERMIT_NUMBER = "M-66666"
     static at = {activeTab=='Permit'}
     static content = {
         activeTab (wait:true) {$("div.ant-tabs-tab-active").text()}
         tabSelect (wait:true) {$("div.ant-tabs-tab", text: "Permit")}
 
         permit_no (wait:true) {$("td", text: contains("BLAHPER-01"), 0)}
+
+        //Permit Tab objects
+        newPermitButton (wait:true) {$("button", text: "Add a New Permit")}
+        newPermitForm { module Form_CreatePermit }
+        permitRow (wait:true) {$("tr").has("div", title: "Permit No.",text: PERMIT_NUMBER)}
+        permitTitle (wait:true) {$("div", title: "Permit No.",text: PERMIT_NUMBER)}
+        addEditButton (wait:true) {$("button").has("div", text: "Add/Edit")}
+        
+        hoverDropdown (wait:true) {$("ul", role:"menu").has("button", text: "Edit permit status").findAll { it.displayed } }
+        editPermitStatusButton (wait:true) {$("button", text: "Edit permit status").findAll { it.displayed } }
+        addAmendmentButton (wait:true) {$("button", text: "Add permit amendment").findAll { it.displayed } }
+        amalgamatePermitButton (wait:true) {$("button", text: "Amalgamate permit").findAll { it.displayed } }
+
+        editPermitFormStatusDropdown (wait:true) {$("div", id: "permit_status_code")}
+        closedDropdownOption (wait:true) {$("li", text: "Closed")}
+        submitEditPermitStatus (wait:true) {$("button", type: "submit")}
+        amendmentDescriptions (wait:true) {$("div", title: "Description")}
+        amendmentDescriptionSpecific (wait:true) {$("div", title: "Description").has(text:"A fancy description" )}
```

Comment:
Suggest moving expected values as a result of user input to constants :)

Reply:
I moved the variables into the comment fields, but left the button names as is.